src/main/java/org/folio/circulation/services/FeeFineService.java

```diff
@@ -1,58 +1,164 @@
 package org.folio.circulation.services;
 
 import static java.util.concurrent.CompletableFuture.allOf;
+import static java.util.concurrent.CompletableFuture.completedFuture;
+import static org.folio.circulation.services.feefine.AccountRefundProcessors.getProcessor;
+import static org.folio.circulation.support.Result.combined;
+import static org.folio.circulation.support.Result.failed;
+import static org.folio.circulation.support.Result.succeeded;
 
 import java.util.Collection;
+import java.util.List;
 import java.util.concurrent.CompletableFuture;
 
+import org.apache.commons.lang3.tuple.ImmutablePair;
+import org.apache.commons.lang3.tuple.Pair;
+import org.folio.circulation.domain.Account;
 import org.folio.circulation.domain.AccountRepository;
-import org.folio.circulation.domain.FeeFineAccountAndAction;
 import org.folio.circulation.domain.FeeFineActionRepository;
-import org.folio.circulation.domain.representations.AccountStorageRepresentation;
-import org.folio.circulation.domain.representations.FeeFineActionStorageRepresentation;
+import org.folio.circulation.domain.ServicePoint;
+import org.folio.circulation.domain.ServicePointRepository;
+import org.folio.circulation.domain.User;
+import org.folio.circulation.domain.UserRepository;
+import org.folio.circulation.domain.representations.StoredAccount;
+import org.folio.circulation.domain.representations.StoredFeeFineAction;
+import org.folio.circulation.services.feefine.AccountRefundContext;
+import org.folio.circulation.services.feefine.AccountRefundProcessor;
+import org.folio.circulation.services.support.CreateAccountCommand;
+import org.folio.circulation.services.support.RefundAccountCommand;
 import org.folio.circulation.support.Clients;
 import org.folio.circulation.support.Result;
 import org.folio.circulation.support.results.CommonFailures;
 
+import io.vertx.core.json.JsonObject;
+
 public class FeeFineService {
```

Comment:
What does this service do? Is it intended to be a facade to the fee / fine API / storage?

Reply:
No, it is not a facade. It encapsulates business logic when dealing with fee/fines. For example, when creating an account, it also creates action, so it will be never forgotten to create. The same think with refund/cancellation.

Reply:
> It encapsulates business logic for dealing with fee/fines. For example, when creating an account, it also creates action, so it will be never forgotten to create. The same think with refund/cancellation.

Ok. Let me rephrase what I'm trying to understand.

Is that logic purely within the domain of fees and fines? Can circulation think of the process as `charging a fee` and the side effects of creating an account and action be considered part of a conversion to the fees and fines domain?

I'm trying to think of a better word than service to describe what this class is responsible for. To me, it provides a simplified view of the fees and fines world to the rest of circulation. Does that fit with your understanding, or do you think of it as doing something different?

Reply:
> To me, it provides a simplified view of the fees and fines world to the rest of circulation. 

Yes, indeed.

Reply:
To me, that fits with the idea of a facade, in that the intention is to provide simplified access to the fees and fines domain.

I got the sense that word does not work for you. What could be a more descriptive word than service to share this intent?

Reply:
To me facade it is something that delegates execution to some another class. 

Maybe `FeeFinePaymentProcessing`?

Reply:
> To me facade it is something that just delegates execution to some another class.

Ok, I personally think that is only one way a facade can be implemented, rather than the intent.

> Maybe FeeFinePaymentProcessing?

My concern with that, is that there are already two other processors, and I'm not really sure what that term means.

Maybe there isn't a better term at this point. I think of processor / service / manager etc as mostly equivalent unless processor means something specific to the domain?

Reply:
I'm fine with `FeeFinePaymentManager`, what do you think?

Reply:
> I'm fine with `FeeFinePaymentManager`, what do you think?

I think manager, service or processor (unless it is a domain term) suffixes are equivalent in not providing much (or any) descriptive content. If the choice is between service and manager, I probably choose service by a nose.

Reply:
My personal preference is for facade because I think it adds descriptive value based upon this [definition](https://en.wikipedia.org/wiki/Facade_pattern):  
> a facade is an object that serves as a front-facing interface masking more complex underlying or structural code

> to make a complex subsystem easier to use, a simple interface should be provided for a set of interfaces in the subsystem.

I'm aware that you weren't keen on facade, which makes me less inclined to use it because it is intended to be more specific to improve communication not cause confusion.

Reply:
Hmm

> a facade is an object that serves as a front-facing interface masking more complex underlying or structural code

This make sense then, will rename to facade then.